src/FeedIo/Rule/Media.php

```diff
@@ -46,11 +48,26 @@ public function setProperty(NodeInterface $node, \DOMElement $element) : void
     {
         if ($node instanceof ItemInterface) {
             $media = $node->newMedia();
-            $media->setNodeName($element->nodeName)
-                ->setType($this->getAttributeValue($element, 'type'))
-                ->setUrl($this->getAttributeValue($element, $this->getUrlAttributeName()))
-                ->setLength($this->getAttributeValue($element, 'length'));
+            $media->setNodeName($element->nodeName);
+
+            if ($element->nodeName == "media:group" or $element->nodeName == "media:content") {
```

Comment:
You can reduce complexity here by replacing the `if` with a `switch` / `case`, for instance with 

```php
            switch ($element->nodeName) {
                case 'media:group':
                    $this->initMedia($media, $element);
                    $media->setUrl($this->getChildAttributeValue($element, 'content', 'url', static::MRSS_NAMESPACE));
                    break;
                case 'media:content':
                    $this->initMedia($media, $element);
                    $media->setUrl($this->getAttributeValue($element, "url"));
                    break;
                default:
                    $media
                        ->setType($this->getAttributeValue($element, 'type'))
                        ->setUrl($this->getAttributeValue($element, $this->getUrlAttributeName()))
                        ->setLength($this->getAttributeValue($element, 'length'));
                    break;
            }
``` 

with initMedia() being

```php
    protected function initMedia(MediaInterface $media, \DOMElement $element): void
    {
        $media->setTitle($this->getChildValue($element, 'title', static::MRSS_NAMESPACE));
        $media->setDescription($this->getChildValue($element, 'description', static::MRSS_NAMESPACE));
        $media->setThumbnail($this->getChildAttributeValue($element, 'thumbnail', 'url', static::MRSS_NAMESPACE));
    }
```

Reply:
Done!